src/awscr-s3/client.cr

```diff
@@ -296,5 +296,21 @@ module Awscr::S3
     private def http
       @http
     end
+
+    private def encode(object)
+      String.build do |io|
+        URI.encode(object, io, space_to_plus: true) do |byte|
+          # check for characters we want to encode
+          # otherwise fall back to default implementation
+          # https://github.com/crystal-lang/crystal/blob/5999ae29beacf4cfd54e232ca83c1a46b79f26a5/src/uri/encoding.cr#L70-L72
+          !should_encode?(byte) && (URI.reserved?(byte) || URI.unreserved?(byte))
```

Comment:
The weirdness of this method is because it is supposed to return false if you want the character encoded